In [1]:
#import libraries
import numpy as np
import pandas as pd

In [2]:
arr = np.array([10, 20, 30, 40, 50])

print(arr)
print("\nAverage number", arr.mean())

[10 20 30 40 50]

Average number 30.0


In [3]:
#indexing
print(arr[3])
print(arr[-1])


40
50


In [4]:
#slicing
print(arr[1:4])
print(arr[::2])
print(arr[:7])
print(arr[-1:-4])
print(arr[:])

[20 30 40]
[10 30 50]
[10 20 30 40 50]
[]
[10 20 30 40 50]


In [5]:
#2D indexing
matrix = np.array([
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
])

print(matrix[0, 1])
print(matrix[:, 1])
print(matrix[1, :])
print(matrix[2, 0:2])

2
[2 5 8]
[4 5 6]
[7 8]


In [6]:
#changing 1D array to 2D array
arr2 = np.arange(1, 13)
reshape = arr2.reshape(2, 6)

print("Original array:", arr2)
print("Reshaped array:", reshape)

#flatten to 1d array
flatten = reshape.flatten()
print("Flatten:", flatten)

Original array: [ 1  2  3  4  5  6  7  8  9 10 11 12]
Reshaped array: [[ 1  2  3  4  5  6]
 [ 7  8  9 10 11 12]]
Flatten: [ 1  2  3  4  5  6  7  8  9 10 11 12]


In [7]:
#working with mathematical operation on array
arr3 = np.array([1, 2, 3, 4, 5])
arr4 = np.array([10, 20, 30, 40, 50])

#basic math operators
print(arr3 + arr4)
print(arr3 - arr4)
print(arr3 * arr4)
print(arr3 / arr4)

#scaler operator
print("Scelar operator\n")
print(arr4**2)
print(arr4 / 2)

[11 22 33 44 55]
[ -9 -18 -27 -36 -45]
[ 10  40  90 160 250]
[0.1 0.1 0.1 0.1 0.1]
Scelar operator

[ 100  400  900 1600 2500]
[ 5. 10. 15. 20. 25.]


In [8]:
#working with matrix element using (dot product)
matrix_1 = np.array([
    [1, 2],
    [3, 4]
])

matrix_2 = np.array([
    [5, 6],
    [7, 8]
])

dot = np.dot(matrix_1, matrix_2)
print(dot)

multiply = matrix_1 * matrix_2
print(multiply)

[[19 22]
 [43 50]]
[[ 5 12]
 [21 32]]


In [27]:
np.random.seed(42)  # for reproducibility
days = np.arange(1, 31)
step = np.random.randint(3456, 12034, size=30)  

In [28]:
data = pd.DataFrame({
    "Day": days,
    "Step": step
})

data

Day   Step
0     1  10726
1     2   4316
2     3   8846
3     4   8647
4     5   9190
5     6   9721
6     7   3922
7     8   7882
8     9   9034
9    10  11778
10   11   5141
11   12   4225
12   13  10405
13   14   5889
14   15   8767
15   16   8507
16   17   9876
17   18   4640
18   19   8011
19   20   6841
20   21   9852
21   22   6014
22   23  11305
23   24   5503
24   25   6203
25   26   3645
26   27   6190
27   28   6461
28   29   8114
29   30   5355

In [29]:
#basic stats
mean_steps = np.mean(steps)
median_steps = np.median(steps)
min_steps = np.min(steps)
max_steps = np.max(steps)
std_steps = np.std(steps)

In [30]:
goal_days = np.where(steps >= 10000)[0] + 1
goal_days

array([ 1, 10, 13, 23])

In [31]:
weeks = np.array_split(steps, 4)
weekly_totals = [week.sum() for week in weeks]
weekly_avg = [week.mean() for week in weeks]

In [32]:
window = 7
moving_avg = np.convolve(steps, np.ones(window)/window, mode='valid')

In [33]:
lower_bound = mean_steps - 2*std_steps
upper_bound = mean_steps + 2*std_steps
anomalies = np.where((steps < lower_bound) | (steps > upper_bound))[0] + 1

In [16]:
# Print results
print("Basic Stats")
print("Mean:", mean_steps, "Median:", median_steps)
print("Min:", min_steps, "Max:", max_steps, "Std Dev:", std_steps)

print("\nGoal Days (>=10,000 steps):", goal_days)

print("\nWeekly Totals:", weekly_totals)
print("Weekly Averages:", weekly_avg)

print("\nFirst 5 values of 7-day Moving Average:", moving_avg[:5])

print("\nAnomalies (Days too low/high):", anomalies)

Basic Stats
Mean: 7500.2 Median: 7946.5
Min: 3645 Max: 11778 Std Dev: 2309.770715893679

Goal Days (>=10,000 steps): [ 1 10 13 23]

Weekly Totals: [np.int64(63250), np.int64(63746), np.int64(56539), np.int64(41471)]
Weekly Averages: [np.float64(7906.25), np.float64(7968.25), np.float64(8077.0), np.float64(5924.428571428572)]

First 5 values of 7-day Moving Average: [7909.71428571 7503.42857143 8177.42857143 8596.28571429 8095.42857143]

Anomalies (Days too low/high): []


Your average daily steps is 7,500, meaning that on a typical day you walk around seventy-five hundred steps.

The median is 7,946, a bit higher than the mean, which suggests most days are on the more active side.

Your lowest day is 3,645 steps, likely a low-movement day, while your highest day reaches 11,778, marking your most active day of the month.
The standard deviation is 2,309, showing that your activity varies a lot — some days are very active, others much lighter.